includes/class-wc-query.php

```diff
@@ -487,12 +505,38 @@ public function product_query( $q ) {
 		self::$product_query = $q;
 
 		// Additonal hooks to change WP Query.
-		add_filter( 'posts_clauses', array( $this, 'price_filter_post_clauses' ), 10, 2 );
+		add_filter(
+			'posts_clauses',
+			function( $args, $wp_query ) {
+				return $this->product_query_post_clauses( $args, $wp_query );
+			},
```

Comment:
Yes, this is to keep the method private. Public methods are a maintenance burden, and as I see it `@internal` is just a workaround and doesn't really guarantee that nobody will go and use the method from an extension (yes, that would be "their fault", but still: more work and headaches for us).

Also we can always change it to `public` in the future if we find that we really need it without breaking compatibility, so I would stick to the "give methods as little visibility as possible" principle here.

Reply:
Understood ✅ 

As a sidebar, it could make for a good discussion re our own coding standards (as to whether we should use this pattern more widely, or avoid its use). For want of a better term, it just doesn't feel very WordPressy to me (and actually the WP coding standards state that [anon functions should not be used in this way](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#closures-anonymous-functions)). Also worth bearing in mind that besides making it harder to unhook, it doesn't actually prevent 3PDs from using the proxy function to call the private method outside of the intended context, there's just an extra level of indirection.

Reply:
>  it doesn't actually prevent 3PDs from using the proxy function to call the private method outside of the intended context, there's just an extra level of indirection.

Yes. But who's at fault if you steal the radio from my car? If I had left the window open we can argue that it's at least partly my fault. But if you broke the window, then it's definitely 100% your fault. 🙂

In the end it's a matter of documentation/intent. `private` clearly signals "you are not supposed to use this". `public` with `@internal` is ok if there's really no other option, but it opens the door to "ooops I just saw it was public so I used it". And whoever wants to purposedly do bad things will do them no matter what.